VictoriaAlbert/VictoriaAlbert/VictoriaObject.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  VictoriaObject.swift
+//  VictoriaAlbert
+//
+//  Created by Margaret Ikeda on 11/10/16.
+//  Copyright © 2016 Margaret Ikeda. All rights reserved.
+//
+
+import Foundation
+
+
+internal enum VictoriaObjectModelParseError: Error {
+    case fieldsDictionary, name, dateText, place, title
+}
+
+internal struct VictoriaObject {
+    let name: String
+    let dateText: String
+    let place: String
+    let title: String?
+//    let imageUrl: String
+//    let thumbUrl: String
+    
+    static func victoriaObjects(from data: Data) -> [VictoriaObject]? {
+        var victoriaObjectsToReturn:[VictoriaObject]? = []
+        
+        do {
+            let jsonData: Any = try JSONSerialization.jsonObject(with: data, options: [])
+            
+            guard let response = jsonData as? [String: Any],
```

Comment:
You were very close to getting this working. You recognized the type of the data as being ```[[String:Any]]``` but made your initial cast without the array so it didn't enter. This is a case where stepping through with the debugger might have been helpful, showing where exactly in this chain of guard/lets the code was dying. Here's a working block based on your code with minimal changes:

```swift
            guard let response = jsonData as? [String: Any],
                let arrayOfDictionaries = response["records"] as? [[String: Any]],
                //let arrayOfDictionaries = records as? [[String: Any]],
                let finalDictionary = arrayOfDictionaries[0] as? [String: Any],
                let fieldsDictionary = finalDictionary["fields"] as? [String: Any]
                else {
                    throw VictoriaObjectModelParseError.fieldsDictionary
            }
            print("@@@@Got fieldsDictionary@@@@")
            
            for victoriaObjectResults in arrayOfDictionaries {
                guard let fieldsDictionary = victoriaObjectResults["fields"] as? [String: Any] else {throw VictoriaObjectModelParseError.fieldsDictionary}

                
                guard let name = fieldsDictionary["object"] as? String
.
.
.
etc.
```